pkg/machine/e2e/start_test.go

```diff
@@ -88,6 +92,69 @@ var _ = Describe("podman machine start", func() {
 		Expect(startSession.errorToString()).To(ContainSubstring("VM already running or starting"))
 	})
 
+	It("start machine with conflict on SSH port", func() {
+		skipIfNotVmtype(define.QemuVirt, "Test should only run in linux")
+
+		i := new(initMachine)
+		session, err := mb.setCmd(i.withImage(mb.imagePath)).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(session).To(Exit(0))
+
+		inspect := new(inspectMachine)
+		inspectSession, err := mb.setCmd(inspect.withFormat("{{.SSHConfig.Port}}")).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(inspectSession).To(Exit(0))
+		inspectPort := inspectSession.outputToString()
+
+		connections := new(listSystemConnection)
+		connectionsSession, err := mb.setCmd(connections.withFormat("{{.URI}}")).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(connectionsSession).To(Exit(0))
+		connectionURLs := connectionsSession.outputToStringSlice()
+		connectionPorts, err := mapToPort(connectionURLs)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(connectionPorts).To(HaveEach(inspectPort))
+
+		ncListenCmdArgs := []string{"-l", "-p", inspectPort}
+		GinkgoWriter.Printf("Running: nc %s\n", strings.Join(ncListenCmdArgs, " "))
+		ncListenCmd := exec.Command("nc", ncListenCmdArgs...)
+		ncListenSession, err := Start(ncListenCmd, GinkgoWriter, GinkgoWriter)
+		Expect(err).ToNot(HaveOccurred())
+
+		time.Sleep(5 * time.Second)
```

Comment:
Sorry I saw the race in the prevous run but slowing down CI like that is not good

Now that I have slept over it I think my ncat suggestion was just stupid
```
listener, err = net.Listen("tcp", "127.0.0.1:" + port)
Expect(err).ToNot(HaveOccurred())
defer listener.Close()
```

That should work on all hypervisors so we can remove the skip, it is race free, simpler and faster